apiserver/facades/client/application/deployrepository.go

```diff
@@ -332,10 +337,12 @@ func makeDeployFromRepositoryValidator(ctx context.Context, cfg validatorConfig)
 }
 
 type deployFromRepositoryValidator struct {
-	model             Model
-	cloudService      common.CloudService
-	credentialService common.CredentialService
-	state             DeployFromRepositoryState
+	model              Model
```

Comment:
Can this usage go ?

Reply:
nope, this is also needed to pass into `assertCharmAssumptions` and then into `stateenvirons.SupportedFeatures`